### PR TITLE
first todoshka

### PR DESCRIPTION
**Pull request by letrylesha**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -3,8 +3,57 @@ import ReactDOM from 'react-dom';
...
+  {id: '002', task: 'Learn Redux'},
+  {id: '003', task: 'Learn Router'},
+  {id: '004', task: 'Learn Grid'},
+  {id: '005', task: 'Learn sOME'},
+  {id: '006', task: 'Learn Router'}
+];
+
+function tasksAction(state: TaskInterface[] = initialState, action: any): TaskInterface[] {
+  if (action.type === 'ADD_TASK') {
+    return [
```
Comment (didevgen) [inline src/index.tsx]: you should always return object

File: src/index.tsx [hunk 2]
```diff
@@ -3,8 +3,57 @@ import ReactDOM from 'react-dom';
...
+      ...state,
+      action.payload,
+    ];
+  }
+  if (action.type === 'DELETE_TASK') {
```
Comment (didevgen) [inline src/index.tsx]: switch case

File: src/index.tsx [hunk 3]
```diff
@@ -3,8 +3,57 @@ import ReactDOM from 'react-dom';
...
+    return [
+      ...filteredTasks
+    ];
+  }
+  if (action.type === 'EDIT_TASK') {
+    state.map( (task, i) => {
+      if (action.payload.id === task.id) {
+        state[i].task = action.payload.update;
+      }
+    });
```
Comment (didevgen) [inline src/index.tsx]: side effect is a bad practise

File: src/index.tsx [hunk 4]
```diff
@@ -3,8 +3,57 @@ import ReactDOM from 'react-dom';
...
+    console.log('ddd');
+    return [];
+  }
+  return state;
+}
+
+const store = createStore(tasksAction);
+
+ReactDOM.render(
+  <Provider store={ store }>
```
Comment (didevgen) [inline src/index.tsx]: you should split it by files

File: src/index.tsx (hunk 1 above)
Comment (letrylesha) [reply src/index.tsx]: fixed

File: src/index.tsx (hunk 2 above)
Comment (letrylesha) [reply src/index.tsx]: fixed

File: src/index.tsx (hunk 4 above)
Comment (letrylesha) [reply src/index.tsx]: fixed

**Pull request merged**